torch/csrc/jit/operator_upgraders/upgraders.h

```diff
@@ -0,0 +1,25 @@
+#pragma once
+#include <c10/macros/Export.h>
+#include <iostream>
+#include <mutex>
+#include <string>
+#include <unordered_map>
+
+namespace torch {
+namespace jit {
+
+struct UpgradersMap {
+    static std::unordered_map<std::string, std::string> content;
+    static std::mutex *lock;
```

Comment:
any reason for pointer instead of just `std::mutex lock;`

Reply:
+1
Also, we should probably use `shared_mutex` so read will be cheaper:
https://en.cppreference.com/w/cpp/thread/shared_mutex